### DISCUSSION TITLE
XML-RPC-API hängt beim zugriff auf port 2001 (http get)

### DISCUSSION THREAD

**Discussion by sven15711** [OpenCCU/OpenCCU#2652] [🇩🇪 Fragen&Antworten] [answered]
Hallo Community,

Ich habe eine frische RaspberryMatic-Installation 3.73.9.20240130 auf einem RPi 3B+ durchgeführt. Dort habe ich 3 Heizthermostate und 5 Fenstersensoren eingerichtet. 

Nun wollte ich den IOBroker anschließen, jedoch kommen keine Daten über die XML-RPC-API von der RaspberryMatic. Wenn ich direkt per curl Port 2000 abfrage, erhalte ich eine HTML-Seite zurück (komplett blau und in der Mitte steht "${ccuNotReadyHint}"). 
Wenn ich Port 2001 mit curl abfrage, hängt der Zugriff. 
Laut curl wird der HTTP-GET-Befehl an RaspberryMatic gesendet, aber es kommt nichts zurück, und es endet in einem Timeout. Die Firewall ist komplett offen, und die Autorisierung habe ich deaktiviert.

![pi_settings1](https://github.com/jens-maus/RaspberryMatic/assets/105542509/e64f1dfc-b6c1-4ff0-8f97-a4e2e1351d6a)
![pi_settings2_firewall](https://github.com/jens-maus/RaspberryMatic/assets/105542509/51d7bec4-4072-401c-887f-2491c9938ec4)

Curl Ausgabe:
```
C:\Users\me>curl -vvvvv http://192.168.178.115:2001
*   Trying 192.168.178.115:2001...
* Connected to 192.168.178.115 (192.168.178.115) port 2001
> GET / HTTP/1.1
> Host: 192.168.178.115:2001
> User-Agent: curl/8.4.0
> Accept: */*
>

```

Was mache ich falsch und welche Settings muss ich noch anpassen?
Vielen Dank und Gruß
Sven

**Comment by jens-maus** [comment]
@sven15711 Was war/ist der Fehler auf deiner Seite gewesen? Bitte darlegen damit auch andere davon lernen können die über diesen Beitrag hie dann ggf. stolpern...

**Reply by sven15711** [reply, marked as answer]
Ja. Es war ein Verständnisproblem auf meiner Seite: 

Nach meinem aktuellen Verständnis kommuniziert IOBroker zu Homematic nicht einfach über das HTTP GET/POST-Verfahren, sondern nutzt XML-RPC und erwartet somit mehr beim Aufruf (siehe [Homematic XML-RPC API Dokumentation](https://www.eq-3.de/Downloads/eq3/download%20bereich/hm_web_ui_doku/HM_XmlRpc_API.pdf)). Wenn ich im Browser oder per curl einfach nur die URL http://<IP>:2001 aufrufe, wartet Homematic noch auf weitere Informationen. Deshalb hängt das Ganze bzw. es passiert nichts.

Im ioBroker musste ich zusätzlich zur Homematic-IP und dem Port 2001 auch eine Callback-Adresse (Docker Host IP) angeben. Da mein ioBroker in einem Docker-Container läuft, musste ich einen zusätzlichen Port 2001 vom Host in den Container weiterleiten, damit das Event-System von Homematic die richtige Adresse+Port des ioBrokers erreichen kann. Ohne die Angabe der Callback-Adresse übermittelt ioBroker die interne IP des Containers, und diese – sowie der Port – sind von außen nicht erreichbar.
